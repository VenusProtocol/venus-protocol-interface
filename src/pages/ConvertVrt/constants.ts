import { getToken } from 'utilities';

export const VRT_ID = 'vrt';
export const VRT_DECIMAL = getToken(VRT_ID).decimals;
