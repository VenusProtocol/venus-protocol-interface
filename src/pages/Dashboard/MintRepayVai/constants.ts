export const VAI_SYMBOL = 'vai';
