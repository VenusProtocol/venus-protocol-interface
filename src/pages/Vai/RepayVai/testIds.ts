const TEST_IDS = {
  repayTextField: 'repay-text-field',
};

export default TEST_IDS;
