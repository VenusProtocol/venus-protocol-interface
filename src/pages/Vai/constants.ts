export const VAI_ID = 'vai';
