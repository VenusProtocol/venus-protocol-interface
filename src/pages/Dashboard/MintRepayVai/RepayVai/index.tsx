/** @jsxImportSource @emotion/react */
import React, { useContext } from 'react';
import BigNumber from 'bignumber.js';
import type { TransactionReceipt } from 'web3-core';

import { convertTokensToWei, convertWeiToTokens } from 'utilities';
import { VError } from 'errors';
import { AmountForm, IAmountFormProps } from 'containers/AmountForm';
import { AuthContext } from 'context/AuthContext';
import useHandleTransactionMutation from 'hooks/useHandleTransactionMutation';
import {
  ConnectWallet,
  EnableToken,
  FormikSubmitButton,
  LabeledInlineContent,
  FormikTokenTextField,
} from 'components';
import { useVaiUser } from 'hooks/useVaiUser';
import { useRepayVai } from 'clients/api';
import { useTranslation } from 'translation';
import useConvertWeiToReadableTokenString from 'hooks/useConvertWeiToReadableTokenString';
import { VAI_ID } from '../constants';
import { useStyles } from '../styles';

export interface IRepayVaiUiProps {
  disabled: boolean;
  isRepayVaiLoading: boolean;
  repayVai: (amountWei: BigNumber) => Promise<TransactionReceipt | undefined>;
  userBalanceWei?: BigNumber;
  userMintedWei?: BigNumber;
}

export const RepayVaiUi: React.FC<IRepayVaiUiProps> = ({
  disabled,
  userBalanceWei,
  userMintedWei,
  isRepayVaiLoading,
  repayVai,
}) => {
  const styles = useStyles();
  const { t } = useTranslation();

  const handleTransactionMutation = useHandleTransactionMutation();

  const limitTokens = React.useMemo(() => {
    const limitWei =
      userBalanceWei && userMintedWei
        ? BigNumber.minimum(userBalanceWei, userMintedWei)
        : new BigNumber(0);

    return convertWeiToTokens({ valueWei: limitWei, tokenId: VAI_ID }).toFixed();
  }, [userBalanceWei?.toFixed(), userMintedWei?.toFixed()]);

  // Convert minted wei into VAI
  const readableRepayableVai = useConvertWeiToReadableTokenString({
    valueWei: userMintedWei,
    tokenId: VAI_ID,
  });

  const hasRepayableVai = userMintedWei?.isGreaterThan(0) || false;

  const onSubmit: IAmountFormProps['onSubmit'] = async amountTokens => {
    const amountWei = convertTokensToWei({
      value: new BigNumber(amountTokens),
      tokenId: VAI_ID,
    });

    return handleTransactionMutation({
      mutate: () => repayVai(amountWei),
      successTransactionModalProps: transactionReceipt => ({
        title: t('mintRepayVai.repayVai.successfulTransactionModal.title'),
        content: t('mintRepayVai.repayVai.successfulTransactionModal.message'),
        amount: {
          valueWei: amountWei,
          tokenId: 'vai',
        },
        transactionHash: transactionReceipt.transactionHash,
      }),
    });
  };

  return (
    <ConnectWallet message={t('mintRepayVai.repayVai.connectWallet')}>
      <EnableToken title={t('mintRepayVai.repayVai.enableToken')} vTokenId={VAI_ID}>
        <AmountForm onSubmit={onSubmit} css={styles.tabContentContainer}>
          {() => (
            <>
              <div css={styles.ctaContainer}>
                <FormikTokenTextField
                  name="amount"
                  css={styles.textField}
                  tokenId={VAI_ID}
                  max={limitTokens}
                  disabled={disabled || isRepayVaiLoading || !hasRepayableVai}
                  rightMaxButton={{
                    label: t('mintRepayVai.repayVai.rightMaxButtonLabel'),
                    valueOnClick: limitTokens,
                  }}
                />

                <LabeledInlineContent
                  css={styles.getRow({ isLast: true })}
                  iconName={VAI_ID}
                  label={t('mintRepayVai.repayVai.repayVaiBalance')}
                >
                  {readableRepayableVai}
                </LabeledInlineContent>
              </div>

              <FormikSubmitButton
                variant="secondary"
                loading={isRepayVaiLoading}
                disabled={disabled}
                enabledLabel={t('mintRepayVai.repayVai.btnRepayVai')}
                fullWidth
              />
            </>
          )}
        </AmountForm>
      </EnableToken>
    </ConnectWallet>
  );
};

const RepayVai: React.FC = () => {
  const { account } = useContext(AuthContext);
  const { userVaiMinted, userVaiBalance } = useVaiUser();

  const { mutateAsync: contractRepayVai, isLoading: isRepayVaiLoading } = useRepayVai();

  // Convert minted VAI balance into wei of VAI
  const userMintedWei = React.useMemo(
    () => convertTokensToWei({ value: userVaiMinted, tokenId: VAI_ID }),
    [userVaiMinted.toFixed()],
  );

  // Convert user VAI balance into wei of VAI
  const userBalanceWei = React.useMemo(
    () => convertTokensToWei({ value: userVaiBalance, tokenId: VAI_ID }),
    [userVaiBalance.toFixed()],
  );

  const repayVai: IRepayVaiUiProps['repayVai'] = async amountWei => {
    if (!account) {
      // This error should never happen, since the form inside the UI component
      // is disabled if there's no logged in account
      throw new VError({ type: 'unexpected', code: 'undefinedAccountErrorMessage' });
    }
    return contractRepayVai({
      fromAccountAddress: account.address,
      amountWei: amountWei.toFixed(),
    });
  };

  return (
    <RepayVaiUi
      disabled={!account}
      userBalanceWei={userBalanceWei}
      userMintedWei={userMintedWei}
      isRepayVaiLoading={isRepayVaiLoading}
      repayVai={repayVai}
    />
  );
};

export default RepayVai;
