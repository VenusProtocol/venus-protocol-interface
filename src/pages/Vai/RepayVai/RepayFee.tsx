/** @jsxImportSource @emotion/react */
import BigNumber from 'bignumber.js';
import { LabeledInlineContent } from 'components';
import React, { useContext, useMemo } from 'react';
import { useTranslation } from 'translation';
import { convertTokensToWei, convertWeiToTokens } from 'utilities';

import { useGetVaiCalculateRepayAmount } from 'clients/api';
import { TOKENS } from 'constants/tokens';
import { AuthContext } from 'context/AuthContext';
import useDebounceValue from 'hooks/useDebounceValue';

import { useStyles } from '../styles';

const DEBOUNCE_DELAY = 300;

export interface IRepayFeeProps {
  repayAmountTokens: string;
}

const RepayFee = ({ repayAmountTokens }: IRepayFeeProps) => {
  const { t } = useTranslation();
  const styles = useStyles();
  const { accountAddress } = useContext(AuthContext);

  const debouncedAmountTokens = useDebounceValue(repayAmountTokens, DEBOUNCE_DELAY);

  const { data: vaiRepayAmountData } = useGetVaiCalculateRepayAmount(
    {
      accountAddress: accountAddress || '',
      repayAmountWei: convertTokensToWei({
        value: new BigNumber(debouncedAmountTokens || 0),
        token: TOKENS.vai,
      }),
    },
    {
      enabled: !!accountAddress,
      keepPreviousData: true,
    },
  );

  const readableRepayFee = useMemo(() => {
    const repayFeeWei = new BigNumber(vaiRepayAmountData?.vaiCurrentInterestWei || 0).plus(
      vaiRepayAmountData?.vaiPastInterestWei || 0,
    );

    const fee = convertWeiToTokens({
      valueWei: repayFeeWei,
      token: TOKENS.vai,

      returnInReadableFormat: true,
    });

    const feePercentage = vaiRepayAmountData?.feePercentage || 0;
    const formattedFeePercentage = `${new BigNumber(feePercentage)
      .dp(feePercentage < 0.01 ? 6 : 2)
      .toNumber()}%`;

    return `${fee} (${formattedFeePercentage})`;
  }, [vaiRepayAmountData?.feePercentage]);

  return (
    <LabeledInlineContent
      css={styles.getRow({ isLast: true })}
      iconSrc="fee"
      label={t('vai.repayVai.repayFeeLabel')}
      tooltip={t('vai.repayVai.repayFeeTooltip')}
    >
      {readableRepayFee}
    </LabeledInlineContent>
  );
};

export default RepayFee;
