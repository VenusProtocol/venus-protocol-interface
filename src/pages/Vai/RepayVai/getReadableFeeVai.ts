import BigNumber from 'bignumber.js';
import { convertWeiToTokens } from 'utilities';

import { TOKENS } from 'constants/tokens';

const getReadableFeeVai = ({
  valueWei,
  feePercentage,
}: {
  valueWei: BigNumber;
  feePercentage: number;
}) => {
  const feeWei = valueWei.multipliedBy(feePercentage).dividedBy(100);
  return convertWeiToTokens({
    valueWei: feeWei,
    token: TOKENS.vai,
    returnInReadableFormat: true,
  });
};

export default getReadableFeeVai;
