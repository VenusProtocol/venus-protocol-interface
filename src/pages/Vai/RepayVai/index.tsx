/** @jsxImportSource @emotion/react */
import BigNumber from 'bignumber.js';
import {
  ApproveToken,
  ConnectWallet,
  FormikSubmitButton,
  LabeledInlineContent,
  NoticeWarning,
  Spinner,
} from 'components';
import { ContractReceipt } from 'ethers';
import React from 'react';
import { useTranslation } from 'translation';
import { convertTokensToWei, convertWeiToTokens, getContractAddress } from 'utilities';

import {
  useGetBalanceOf,
  useGetMintedVai,
  useGetVaiRepayAmountWithInterests,
  useRepayVai,
} from 'clients/api';
import { DEFAULT_REFETCH_INTERVAL_MS } from 'constants/defaultRefetchInterval';
import MAX_UINT256 from 'constants/maxUint256';
import { TOKENS } from 'constants/tokens';
import { AmountForm, AmountFormProps } from 'containers/AmountForm';
import { useAuth } from 'context/AuthContext';
import useConvertWeiToReadableTokenString from 'hooks/useConvertWeiToReadableTokenString';
import useHandleTransactionMutation from 'hooks/useHandleTransactionMutation';

import FormikTokenTextFieldWithBalance from '../TextFieldWithBalance';
import { useStyles } from '../styles';
import RepayFee from './RepayFee';

const vaiControllerContractAddress = getContractAddress('vaiController');

export interface IRepayVaiUiProps {
  disabled: boolean;
  isInitialLoading: boolean;
  isSubmitting: boolean;
  repayVai: (amountWei: BigNumber) => Promise<ContractReceipt | undefined>;
  userBalanceWei?: BigNumber;
  repayBalanceWei?: BigNumber;
  userMintedWei?: BigNumber;
}

const isPayingFullRepayBalance = (
  amount: string,
  repayBalanceWei: BigNumber | undefined,
): boolean => {
  if (amount && repayBalanceWei) {
    const amountWei = convertTokensToWei({ value: new BigNumber(amount), token: TOKENS.vai });
    return amountWei.isEqualTo(repayBalanceWei);
  }

  return false;
};

export const RepayVaiUi: React.FC<IRepayVaiUiProps> = ({
  disabled,
  userBalanceWei,
  repayBalanceWei,
  userMintedWei,
  isInitialLoading,
  isSubmitting,
  repayVai,
}) => {
  const styles = useStyles();
  const { t, Trans } = useTranslation();

  const handleTransactionMutation = useHandleTransactionMutation();

  const limitTokens = React.useMemo(() => {
    const limitWei =
      userBalanceWei && userMintedWei
        ? BigNumber.minimum(userBalanceWei, userMintedWei)
        : new BigNumber(0);

    return convertWeiToTokens({ valueWei: limitWei, token: TOKENS.vai }).toFixed();
  }, [userBalanceWei?.toFixed(), userMintedWei?.toFixed()]);

  // Convert repay balance (minted + interests) into VAI
  const readableRepayableVai = useConvertWeiToReadableTokenString({
    valueWei: repayBalanceWei,
    token: TOKENS.vai,
  });

  const hasRepayableVai = userMintedWei?.isGreaterThan(0) || false;

  const onSubmit: AmountFormProps['onSubmit'] = async amountTokens => {
    const amountWei = convertTokensToWei({
      value: new BigNumber(amountTokens),
      token: TOKENS.vai,
    });

    return handleTransactionMutation({
      mutate: () => repayVai(amountWei),
      successTransactionModalProps: contractReceipt => ({
        title: t('vai.repayVai.successfulTransactionModal.title'),
        content: t('vai.repayVai.successfulTransactionModal.message'),
        amount: {
          valueWei: amountWei,
          token: TOKENS.vai,
        },
        transactionHash: contractReceipt.transactionHash,
      }),
    });
  };

  return (
    <ConnectWallet message={t('vai.repayVai.connectWallet')}>
      <ApproveToken
        title={t('vai.repayVai.approveToken')}
        token={TOKENS.vai}
        spenderAddress={vaiControllerContractAddress}
      >
        {isInitialLoading ? (
          <Spinner />
        ) : (
          <AmountForm onSubmit={onSubmit} css={styles.tabContentContainer}>
            {({ values }) => (
              <>
                <FormikTokenTextFieldWithBalance
                  disabled={disabled || isSubmitting || !hasRepayableVai}
                  maxValue={limitTokens}
                  userBalanceWei={userBalanceWei}
                  maxButtonLabel={t('vai.repayVai.rightMaxButtonLabel')}
                />

                <div css={styles.ctaContainer}>
                  <LabeledInlineContent
                    css={styles.getRow({ isLast: false })}
                    iconSrc={TOKENS.vai}
                    label={t('vai.repayVai.repayVaiBalance')}
                    tooltip={t('vai.repayVai.repayVaiBalanceTooltip')}
                  >
                    {readableRepayableVai}
                  </LabeledInlineContent>

                  <RepayFee repayAmountTokens={values.amount} />
                </div>

                {isPayingFullRepayBalance(values.amount, repayBalanceWei) && (
                  <NoticeWarning
                    css={styles.noticeWarning}
                    description={<Trans i18nKey="vai.repayVai.fullRepayWarning" />}
                  />
                )}

                <FormikSubmitButton
                  loading={isSubmitting}
                  disabled={disabled}
                  enabledLabel={t('vai.repayVai.submitButtonLabel')}
                  disabledLabel={t('vai.repayVai.submitButtonDisabledLabel')}
                  fullWidth
                />
              </>
            )}
          </AmountForm>
        )}
      </ApproveToken>
    </ConnectWallet>
  );
};

const RepayVai: React.FC = () => {
  const { accountAddress } = useAuth();
  const { data: mintedVaiData, isLoading: isGetMintedVaiLoading } = useGetMintedVai(
    {
      accountAddress,
    },
    {
      enabled: !!accountAddress,
    },
  );

  const { data: repayAmountWithInterests, isLoading: isGetVaiRepayAmountWithInterests } =
    useGetVaiRepayAmountWithInterests(
      {
        accountAddress,
      },
      {
        enabled: !!accountAddress,
      },
    );

  const { data: userVaiBalanceData, isLoading: isGetUserVaiBalance } = useGetBalanceOf(
    {
      accountAddress: accountAddress || '',
      token: TOKENS.vai,
    },
    {
      enabled: !!accountAddress,
      refetchInterval: DEFAULT_REFETCH_INTERVAL_MS,
    },
  );

  const { mutateAsync: contractRepayVai, isLoading: isSubmitting } = useRepayVai();

  const repayVai: IRepayVaiUiProps['repayVai'] = async amountWei => {
    const isRepayingFullLoan = amountWei.eq(
      convertTokensToWei({
        value: repayAmountWithInterests!.vaiRepayAmountWithInterests,
        token: TOKENS.vai,
      }),
    );

    return contractRepayVai({
      amountWei: isRepayingFullLoan ? MAX_UINT256 : amountWei,
    });
  };

  return (
    <RepayVaiUi
      disabled={!accountAddress}
      userBalanceWei={userVaiBalanceData?.balanceWei}
      repayBalanceWei={repayAmountWithInterests?.vaiRepayAmountWithInterests}
      userMintedWei={mintedVaiData?.mintedVaiWei}
      isInitialLoading={
        isGetMintedVaiLoading || isGetUserVaiBalance || isGetVaiRepayAmountWithInterests
      }
      isSubmitting={isSubmitting}
      repayVai={repayVai}
    />
  );
};

export default RepayVai;
