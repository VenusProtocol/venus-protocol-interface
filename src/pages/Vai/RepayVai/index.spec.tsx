import { fireEvent, waitFor } from '@testing-library/react';
import BigNumber from 'bignumber.js';
import React from 'react';
import { convertTokensToWei, convertWeiToTokens } from 'utilities';

import fakeMulticallResponses from '__mocks__/contracts/multicall';
import fakeAccountAddress from '__mocks__/models/address';
import fakeContractReceipt from '__mocks__/models/contractReceipt';
import {
  getBalanceOf,
  getMintedVai,
  getVaiCalculateRepayAmount,
  getVaiRepayAmountWithInterests,
  repayVai,
} from 'clients/api';
import formatToOutput from 'clients/api/queries/getVaiCalculateRepayAmount/formatToOutput';
import formatToGetVaiRepayAmountWithInterestsOutput from 'clients/api/queries/getVaiRepayAmountWithInterests/formatToOutput';
import { TOKENS } from 'constants/tokens';
import useSuccessfulTransactionModal from 'hooks/useSuccessfulTransactionModal';
import renderComponent from 'testUtils/renderComponent';
import en from 'translation/translations/en.json';

import RepayVai from '.';
import TEST_IDS from '../testIds';

jest.mock('clients/api');
jest.mock('components/Toast');
jest.mock('hooks/useSuccessfulTransactionModal');

jest.useFakeTimers();

const fakeUserVaiMintedWei = new BigNumber('100000000000000000000');
const repayInputAmountTokens = '100';

describe('pages/Dashboard/MintRepayVai/RepayVai', () => {
  beforeEach(() => {
    (getMintedVai as jest.Mock).mockImplementation(() => ({
      mintedVaiWei: fakeUserVaiMintedWei,
    }));

    (getVaiRepayAmountWithInterests as jest.Mock).mockImplementation(() =>
      formatToGetVaiRepayAmountWithInterestsOutput({
        contractCallResults: fakeMulticallResponses.vaiController.getVaiRepayTotalAmount,
      }),
    );

    (getVaiCalculateRepayAmount as jest.Mock).mockImplementation(() =>
      formatToOutput({
        repayAmountWei: convertTokensToWei({
          value: new BigNumber(repayInputAmountTokens),
          token: TOKENS.vai,
        }),
        contractCallResults: fakeMulticallResponses.vaiController.getVaiRepayInterests,
      }),
    );
  });

  it('renders without crashing', () => {
    renderComponent(() => <RepayVai />, {
      authContextValue: {
        accountAddress: fakeAccountAddress,
      },
    });
  });

  it('displays the correct repay VAI balance and APY', async () => {
    const { getByText, container } = renderComponent(() => <RepayVai />, {
      authContextValue: {
        accountAddress: fakeAccountAddress,
      },
    });
    await waitFor(() => getByText(en.vai.repayVai.submitButtonDisabledLabel));

    expect(container.textContent).toMatchSnapshot();
  });

  it('lets user repay their VAI balance', async () => {
    const { openSuccessfulTransactionModal } = useSuccessfulTransactionModal();

    (repayVai as jest.Mock).mockImplementationOnce(async () => fakeContractReceipt);

    (getBalanceOf as jest.Mock).mockImplementation(() => ({
      balanceWei: fakeUserVaiMintedWei,
    }));

    const { getByText, getByPlaceholderText } = renderComponent(() => <RepayVai />, {
      authContextValue: {
        accountAddress: fakeAccountAddress,
      },
    });
    await waitFor(() => getByText(en.vai.repayVai.submitButtonDisabledLabel));

    // Input amount
    const fakeUserVaiMinted = convertWeiToTokens({
      valueWei: fakeUserVaiMintedWei,
      token: TOKENS.vai,
    });

    const tokenTextFieldInput = getByPlaceholderText('0.00') as HTMLInputElement;
    fireEvent.change(tokenTextFieldInput, { target: { value: fakeUserVaiMinted.toFixed() } });

    // Check input value updated correctly
    await waitFor(() => expect(tokenTextFieldInput.value).toBe(fakeUserVaiMinted.toFixed()));

    // Submit repayment request
    await waitFor(() => expect(getByText(en.vai.repayVai.submitButtonLabel)));

    const submitButton = getByText(en.vai.repayVai.submitButtonLabel).closest(
      'button',
    ) as HTMLButtonElement;
    fireEvent.click(submitButton);

    // Check repayVai was called correctly
    await waitFor(() => expect(repayVai).toHaveBeenCalledTimes(1));
    expect(repayVai).toHaveBeenCalledWith({
      amountWei: fakeUserVaiMintedWei,
    });

    // Check successful transaction modal is displayed
    await waitFor(() => expect(openSuccessfulTransactionModal).toHaveBeenCalledTimes(1));
    expect(openSuccessfulTransactionModal).toHaveBeenCalledWith({
      transactionHash: fakeContractReceipt.transactionHash,
      amount: {
        token: TOKENS.vai,
        valueWei: fakeUserVaiMintedWei,
      },
      content: expect.any(String),
      title: expect.any(String),
    });
  });

  it('displays the VAI repay fee', async () => {
    const { getByText, getByTestId } = renderComponent(() => <RepayVai />, {
      authContextValue: {
        accountAddress: fakeAccountAddress,
      },
    });

    await waitFor(() => getByText(en.vai.repayVai.repayFeeLabel));

    const tokenTextFieldInput = getByTestId(TEST_IDS.repayTextField) as HTMLInputElement;
    fireEvent.change(tokenTextFieldInput, { target: { value: repayInputAmountTokens } });

    await waitFor(() => expect(tokenTextFieldInput.value).toBe(repayInputAmountTokens));

    // Check user repay VAI balance displays correctly
    await waitFor(() => getByText('0.00032 VAI (0.000317%)'));
  });

  // TODO: add tests to cover failing scenarios (see VEN-631)
});
