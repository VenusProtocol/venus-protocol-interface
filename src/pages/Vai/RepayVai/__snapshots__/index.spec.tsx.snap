// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Dashboard/MintRepayVai/RepayVai displays the correct repay VAI balance and APY 1`] = `"MAXWallet balance: 0 VAIRepay VAI balance5.67M VAIRepay fee0.00032 VAI (0.000317%)Enter a valid amount to repay"`;
